coderd/users.go

```diff
@@ -411,7 +418,8 @@ func (api *API) putUserPassword(rw http.ResponseWriter, r *http.Request) {
 
 	// admins can change passwords without sending old_password
 	if params.OldPassword == "" {
-		if !api.Authorize(rw, r, rbac.ActionUpdate, rbac.ResourceUser.WithID(user.ID.String())) {
+		if !api.Authorize(r, rbac.ActionUpdate, rbac.ResourceUser.WithID(user.ID.String())) {
```

Comment:
as an aside, it is very unintuitive to me that this check should fail for my own user.

Reply:
Why not? There is `ResourceUser` and `ResourceUserData`. The former cannot be edited by "me"

Reply:
I guess that's just it.  Why can't I edit my own "ResourceUser" resource?  The code says

```
	// ResourceUser is the user in the 'users' table.
	// ResourceUser never has any owners or in an org, as it's site wide.
	// 	create/delete = make or delete a new user.
	// 	read = view all 'user' table data
	// 	update = update all 'user' table data
	ResourceUser = Object{
		Type: "user",
	}
```

Things like email and password are in the users table, and users can change them, no?

Reply:
Yea, I need to fix that comment. The comment is incorrect as I since placed `email` as a `ResourceUserData`. There was some thrashing there and now the comment is stale.

My mistake